cmd/run/run.go

```diff
@@ -459,23 +473,25 @@ func RunServer(ctx context.Context, config *Config) error {
 		return fmt.Errorf("failed to initialize authenticator: %w", err)
 	}
 
-	unaryServerInterceptors := []grpc.UnaryServerInterceptor{
+	unaryInterceptors := []grpc.UnaryServerInterceptor{
 		grpc_validator.UnaryServerInterceptor(),
-		grpc_auth.UnaryServerInterceptor(middleware.AuthFunc(authenticator)),
+		otelgrpc.UnaryServerInterceptor(otelgrpc.WithTracerProvider(tp)),
```

Comment:
I'm wondering about the order of these middlewares..

Should we have a `log`, `request_id` and `trace` middleware run before the validation? Even if the request does not meet the protobuf validation requirements, it would be good to have the request id in the log context and the trace info. If validation was taking a long time the trace's would give us a good indication of that and if users are having validation problems we could see the requests and correlate with the request_id.

What's the right approach here? On one hand I don't super chatty logs with request validation problems, but maybe that just means those should fall under a debug level instead of error level.